### DISCUSSION TITLE
Entity Selection

### DISCUSSION THREAD

**Discussion by Quasarrs** [mozman/ezdxf#1092] [Q&A]
1) I started using a very good tool. 
Here's what I noticed.
If you use "bbox_inside" and there is an Mtext in the selection zone that has an insertion point inside the zone and the allowable transfer length "width" is outside the zone, then this object does not fall into the selection zone.
Tell me this is how it should be?

2) I also wanted to know the second small question.
Entities TEXT. Is there a method that returns both the content of the text and its formatting parameters? 
MText has a .text method that outputs something similar.

3) There is such a wonderful method append(text: str)→ MText which adds meaning to an existing text.
Is it possible not just to add but to change the content?

Sorry that everything is written chaotically

**Comment by mozman** [comment]
1. Selects entities whose bounding box lies withing the selection shape. The bounding box of text based entities are calculated from the DXF attributes and they can be wrong.
2. MTEXT supports inline formatting and TEXT supports some control codes but there are no DXF attributes for formatting text entities.
3. Replace the text content `Text.dxf.text == "New text"` same for MTEXT entities.
